### DISCUSSION TITLE
Add clearable to Listbox (Select)

### DISCUSSION THREAD

**Discussion by Trosterud** [tailwindlabs/headlessui#2588] [Ideas]
A X button to clear the selected input. Especially since the Listbox Option { Selected } state doesn't update if you just call onChange within the listbox with either null or undefined as a value.

We ended up not using the Listbox Opetion Selected state at all and do a manual check if the id matches the selectedId to make this

Thank, and have a wonderful day!

**Comment by kripod** [comment]
I think just exposing a `clear` function from within the top-level render props would be a great step forward. It’d set the control’s value to `undefined`, regardless whether it has a controlled state or not.
